client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-layout.tsx

```diff
@@ -1,62 +1,59 @@
-import { Button } from '@automattic/components';
-import { Icon, header, footer, group } from '@wordpress/icons';
-import classnames from 'classnames';
+import { isEnabled } from '@automattic/calypso-config';
+import { Button, Popover } from '@automattic/components';
+import { Icon, plus, header, footer, group } from '@wordpress/icons';
 import { useTranslate } from 'i18n-calypso';
+import { useState, useRef } from 'react';
 import AsyncLoad from 'calypso/components/async-load';
 import PatternActionBar from './pattern-action-bar';
 import type { Pattern } from './types';
 
 type PatternLayoutProps = {
-	header: Pattern | null;
+	header?: Pattern | null;
 	sections: Pattern[];
-	footer: Pattern | null;
-	onAddHeader: () => void;
-	onReplaceHeader: () => void;
-	onDeleteHeader: () => void;
+	footer?: Pattern | null;
+	onAddHeader?: () => void;
+	onReplaceHeader?: () => void;
+	onDeleteHeader?: () => void;
 	onAddSection: () => void;
 	onReplaceSection: ( position: number ) => void;
 	onDeleteSection: ( position: number ) => void;
 	onMoveUpSection: ( position: number ) => void;
 	onMoveDownSection: ( position: number ) => void;
-	onAddFooter: () => void;
-	onReplaceFooter: () => void;
-	onDeleteFooter: () => void;
-	onContinueClick: () => void;
+	onAddFooter?: () => void;
+	onReplaceFooter?: () => void;
+	onDeleteFooter?: () => void;
 };
 
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
 const PatternLayout = ( {
 	header: selectedHeader,
 	sections,
 	footer: selectedFooter,
-	onAddHeader,
-	onReplaceHeader,
-	onDeleteHeader,
+	onAddHeader = noop,
+	onReplaceHeader = noop,
+	onDeleteHeader = noop,
 	onAddSection,
 	onReplaceSection,
 	onDeleteSection,
 	onMoveUpSection,
 	onMoveDownSection,
-	onAddFooter,
-	onReplaceFooter,
-	onDeleteFooter,
-	onContinueClick,
+	onAddFooter = noop,
+	onReplaceFooter = noop,
+	onDeleteFooter = noop,
 }: PatternLayoutProps ) => {
 	const translate = useTranslate();
+	const [ isPopoverVisible, setIsPopoverVisible ] = useState( false );
+	const addButtonContainerRef = useRef< HTMLDivElement | null >( null );
+	const isSidebarRevampEnabled = isEnabled( 'pattern-assembler/sidebar-revamp' );
 
-	return (
-		<div className="pattern-layout">
-			<div className="pattern-layout__header">
-				<h2>{ translate( 'Design your home' ) }</h2>
-				<p>
-					{ translate(
-						'Choose from our library of patterns to quickly put together the structure of your homepage.'
-					) }
-				</p>
-			</div>
-			<div className="pattern-layout__body">
-				<ul>
+	if ( ! isSidebarRevampEnabled ) {
```

Comment:
Display header, sections, and footer for the current layout. It should be the same as before and has some styles change

![image](https://user-images.githubusercontent.com/13596067/220594737-42edefad-45a2-4d11-b4f7-1883d5c7ee6f.png)